internal/tasks/callback_manager_test.go

```diff
@@ -0,0 +1,335 @@
+		callbackManager.FleetSourceUpdated(uuid.New(), "name")
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("DeviceSourceUpdated", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits FleetValidateTask", func() {
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.DeviceSourceUpdated(uuid.New(), "name")
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("RepositoryUpdatedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits RepositoryUpdatesTask", func() {
+		repository := createTestRepository("name", "url")
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.RepositoryUpdatedCallback(repository)
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("AllRepositoriesDeletedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits RepositoryUpdatesTask", func() {
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.AllRepositoriesDeletedCallback(uuid.New())
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("AllFleetsDeletedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits FleetSelectorMatchTask", func() {
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.AllFleetsDeletedCallback(uuid.New())
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("AllDevicesDeletedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits FleetSelectorMatchTask", func() {
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.AllDevicesDeletedCallback(uuid.New())
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("TemplateVersionCreatedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits FleetSelectorMatchTask", func() {
+		templateVersion := createTestTemplateVersion("name", "template")
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.TemplateVersionCreatedCallback(templateVersion)
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+var _ = Describe("TemplateVersionValidatedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	It("submits FleetRolloutTask", func() {
+		templateVersion := createTestTemplateVersion("name", "template")
+		callbackManager := NewCallbackManager(mockPublisher, logger)
+		callbackManager.TemplateVersionValidatedCallback(templateVersion)
+		Expect(mockPublisher.publishCallCount).To(Equal(1))
+	})
+})
+
+func createTestFleet(name string, templateSpec string, selector string) *model.Fleet {
```

Comment:
test/util/create_utils.go has similar methods, except that they write to the DB.  I suggest putting these methods in some common file so that other unit tests can reuse them.